pymeasure/instruments/jobinyvon/jobinyvon270m.py

```diff
@@ -0,0 +1,360 @@
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pyvisa.constants import ControlFlow, Parity, StopBits
+import time
+from pyvisa import VisaIOError
+
+
+def get_steps_returns(steps_str: str):
+    """
+    When asked for the number of steps, the spectrometer returns a string in
+    format "o(number of steps)\r", with this function we get rid of the "o" and
+    the "\r" and we convert the resulting string to an integer.
+    """
+    steps_str = steps_str.lstrip('o')
+    steps_str = steps_str.rstrip('\r')
+    return int(steps_str)
+
+
+def read_int(string: str):
+    """
+    When we expect a numeric answer from the spectrometer, it sometimes sends
+    some more characters in between, and with this function we get rid of those
+    extra characters.
+    """
+    numeric_string = ''.join(char for char in string if char.isdigit())
+    return int(numeric_string)
+
+
+class JY270M(Instrument):
+    """
+    This class represents the Jobin-Yvon 270M Driver.
+    """
+
+    _steps_nm = 32  # Number of motor steps per nm
+    _nm_step = 1 / _steps_nm  # nm per motor step
+    _slit_steps_micron = 0.560  # slit steps per micron
+    _slit_microns_step = 1.0 / _slit_steps_micron  # microns per slit step
+
+    _lambda_max = 1171.68  # Maximum value for the wavelength in nm
+    _max_steps = 37494  # maximum number of steps for the grating motor
+    _max_steps_slit = 1102.36  # do not know what this is for?
+
+    @property
+    def default_timeout(self):
+        """
+        Usual timeout needed for normal use.
+        """
+        return 300
+
+    @property
+    def lambda_0(self):
+        """
+        Get the wavelength when the motor just finished the initialization.
+        """
+        return self._lambda_max
+
+    @property
+    def steps_in_one_nanometer(self):
+        return self._steps_nm
+
+    @property
+    def nanometers_in_one_step(self):
+        return self._nm_step
+
+    @property
+    def micrometers_in_one_step(self):
+        return self._slit_microns_step
+
+    @property
+    def max_grating_steps(self):
+        return self._max_steps
+
+    def __init__(self, adapter, name="JY270M", **kwargs):
+
+        kwargs.update(dict(baud_rate=9600,
+                           timeout=self.default_timeout,
+                           parity=Parity.none,
+                           data_bits=8,
+                           stop_bits=StopBits.one,
+                           flow_control=ControlFlow.dtr_dsr,
+                           write_termination='',
+                           read_termination='',
+                           includeSCPI=False))
+
+        super().__init__(
+            adapter,
+            name,
+            **kwargs)
+
+    gsteps = Instrument.control(
+        'H0\r',
+        'F0,%d\r',
+        "Control the relative step displacement of the grating motor.",
+        get_process=get_steps_returns,
+    )
+
+    entrysteps = Instrument.control(
```

Comment:
Please separate words by underscore:
```suggestion
    entry_steps = Instrument.control(
```